src/hooks/use-media-query.ts

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect } from "react";
+
+const BREAK_POINTS = {
+  sm: "(max-width: 640px)",
+  md: "(max-width: 768px)",
+  lg: "(max-width: 1024px)",
+  xl: "(max-width: 1280px)",
+};
+
+const useMediaQuery = (query: keyof typeof BREAK_POINTS): boolean => {
+  const [matches, setMatches] = useState<boolean>(false);
+
+  useEffect(() => {
```

Comment:
@nadirhussain786  I'll suggest you useMemo and you can impove this code

Reply:
fixed